niceman/resource/ssh.py

```diff
@@ -133,34 +135,36 @@ def _execute_command(self, command, env=None, cwd=None):
 
         # If a command fails, a CommandError exception will be thrown.
         escaped_command = ' '.join(quote(s) for s in command)
-        for i, line in enumerate(self.ssh.execute(escaped_command)):
-            lgr.debug("exec#%i: %s", i, line.rstrip())
+        out = []
+        try:
+            for i, line in enumerate(self.ssh.execute(escaped_command)):
+                out.append(line.rstrip())
+                lgr.debug("exec#%i: %s", i, line.rstrip())
+        except RemoteCommandFailed as e:
+            raise CommandError(cmd=escaped_command, msg=e.msg)
+
+        return (out, None)
 
     def exists(self, path):
         """Return if file exists"""
         return self.ssh.path_exists(path)
 
-    def put(self, src_path, dest_path, preserve_perms=False,
-                owner=None, group=None, recursive=False):
+    def put(self, src_path, dest_path, owner=None, group=None):
         """Take file on the local file system and copy over into the session
         """
         self.ssh.put([src_path], remotepath=dest_path)
 
-    def get(self, src_path, dest_path, preserve_perms=False,
-                  owner=None, group=None, recursive=False):
+        if owner or group:
+            self.chown(owner, group, dest_path, recursive=True)
+
+
+    def get(self, src_path, dest_path, owner=None, group=None):
         """Retrieve a file from the remote system
         """
         self.ssh.get(src_path, localpath=dest_path)
 
-    def chmod(self, mode, remote_path):
-        """Set the mode of a remote path
-        """
-        self.ssh.chmod(mode, remote_path)
-
-    def chown(self, uid, gid, remote_path):
-        """Set the user and group of a path
-        """
-        self.ssh.chown(uid, gid, remote_path)
+        if owner or group:
+            self.chown(owner, group, dest_path, recursive=True)
```

Comment:
again -- this sets permissions on remote end, and not locally

Reply:
when fixed, please add the test to get/put calls which would verify correct operation in this regard